lib/generators/rolemodel/source_map/source_map/app/middleware/rolemodel/source_map_generator.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Rolemodel
+  class SourceMapGenerator
+    def initialize(app, allowed_users_emails = [], **options)
+      root = options.delete(:root) || 'maps'
+      default_headers = {'Set-Cookie' => 'Same-Site=None', 'Cache-Control' => 'max-age=0;no-cache'}
+      custom_headers = default_headers.merge(options.delete(:headers) || {})
+      @app = app
+      @allowed_users_emails = allowed_users_emails
+      @file_server = Rack::Files.new(root, custom_headers)
+    end
+
+    def call(env)
+      serve? ? @file_server.call(env.tap { |env| env['PATH_INFO'].sub!(/^\/[\w-]+[^\/]/, '') }) : @app.call(env)
+    end
+
+    private
+
+    def serve?(env)
+      ENV['RAILS_ENV'] == 'production' && env['PATH_INFO'].match?(/\.map\z/) && permit_user?(env)
+    end
+
+    def permit_user?(env)
+      (current_user_email = current_user(env)&.email) &&
+        (current_user_email.include?('@rolemodelsoftware.com') || @allowed_users_emails.include?(current_user_email))
```

Comment:
I like the blanket permission for `@rolemodelsoftware.com`

Reply:
I added this 'wild card' because our support accounts have inconsistent email prefixes. This check eliminates the manual configuration step. From a security standpoint, this is anti-patter, but most of our apps have only one user with the "@rolemodelsoftware.com" domain (sure, this is a false statement). So, this should not be a problem, but I would like to discuss it.

Reply:
We decided not to go on this path. So, I will remove this condition.

Reply:
On second thought, some of our deployed systems (like LightningCAD DPQ) allow users to register without validating email address ownership. So, blanket allowance for @rolemodelsoftware.com would be a problem.